assets/src/js/bindings/sources.js

```diff
@@ -59,7 +69,7 @@ registerBlockBindingsSource( {
 					}
 
 					result[ attribute ] = value;
-				} else if ( typeof fieldValue === 'number' ) {
+				} else if ( 'number' === typeof fieldValue ) {
```

Comment:
Would it make sense to replace this with

```suggestion
				} else if ( 'number' === fieldType ) {
```

now that we have `fieldType` available?

Reply:
I refactored the entire part 😅